src/eureka/S3_data_reduction/wfc3.py

```diff
@@ -613,130 +806,143 @@ def fit_bg(dataim, datamask, datav0, datavariance, n, meta, isplots=0):
-        for n in range(meta.nreads-1):
-            writeDrift2D(hst.calcDrift2D((meta.subdata_ref[p][0] *
-                                          meta.subdiffmask[p][0]),
-                                         (data.flux[n] *
-                                          meta.subdiffmask[-1][n]),
-                                         m, n))
+        for n in range(data.flux.shape[0]):
+            # Get read number
+            r = n % meta.nreads
+            # Get index of reference frame
+            # (0 = forward scan, 1 = reverse scan)
+            p = data.scandir.values[n]
+            writeDrift2D(hst.calcDrift2D((meta.subdata_ref[p][r] *
+                                          meta.subdiffmask_ref[p][r]),
+                                         (data.flux[n]*data.flatmask[n]),
+                                         n))
     else:
         # Multiple CPUs
         pool = mp.Pool(meta.ncpu)
-        # Get index of reference frame
-        # (0 = forward scan, 1 = reverse scan)
-        p = meta.scandir[m]
-        for n in range(meta.nreads-1):
+        for n in range(data.flux.shape[0]):
+            # Get read number
+            r = n % meta.nreads
+            # Get index of reference frame
+            # (0 = forward scan, 1 = reverse scan)
+            p = data.scandir.values[n]
             res = pool.apply_async(hst.calcDrift2D,
-                                   args=((meta.subdata_ref[p][0] *
-                                          meta.subdiffmask[p][0]),
-                                         (data.flux[n] *
-                                          meta.subdiffmask[-1][n]),
-                                         m, n),
+                                   args=((meta.subdata_ref[p][r] *
+                                          meta.subdiffmask_ref[p][r]),
+                                         (data.flux[n]*data.flatmask[n]),
+                                         n),
                                    callback=writeDrift2D)
         pool.close()
         pool.join()
         res.wait()
 
-    print("Performing rough, pixel-scale drift correction...")
-    meta.drift2D_int.append(np.round(meta.drift2D[-1], 0))
+    data['drift2D'] = xr.DataArray(drift2D, name='drift2D',
+                                   coords={"time": data.time,
+                                           "axis": ["x", "y"]},
+                                   dims=["time", "axis"],
+                                   attrs={"time_units": data.time.time_units,
+                                          "drift_units": 'pixels'})
+    data.drift2D["time"].attrs["time_units"] = data.time.time_units
+
+    log.writelog("  Performing rough, pixel-scale drift correction...",
+                 mute=(not meta.verbose))
+    drift2D_int = np.round(drift2D, 0)
     # Correct for drift by integer pixel numbers, no interpolation
-    for n in range(meta.nreads-1):
+    for n in range(data.flux.shape[0]):
```

Comment:
Let's define `nint = data.flux.shape[0]` once at the top and avoid calling this calculation multiple time.